test/base/test_buffer.py

```diff
@@ -112,9 +115,116 @@ def test_update():
     assert (buf2[-1].obs == buf1[0].obs).all()
 
 
+def test_segtree():
+    for op, init in zip(['sum', 'max', 'min'], [0., -np.inf, np.inf]):
+        realop = getattr(np, op)
+        # small test
+        actual_len = 8
+        tree = SegmentTree(actual_len, op)  # 1-15. 8-15 are leaf nodes
+        assert np.all([tree[i] == init for i in range(actual_len)])
+        with pytest.raises(IndexError):
+            tree[actual_len]
+        naive = np.zeros([actual_len]) + init
+        for _ in range(1000):
+            # random choose a place to perform single update
+            index = np.random.randint(actual_len)
+            value = np.random.rand()
+            naive[index] = value
+            tree[index] = value
+            for i in range(actual_len):
+                for j in range(i, actual_len):
+                    try:
+                        ref = realop(naive[i:j])
+                    except ValueError:
+                        continue
+                    out = tree.reduce(i, j)
+                    assert np.allclose(ref, out), (i, j, ref, out)
```

Comment:
At first glance, I think this is wrong. But later I figure out this is ``assert np.allclose(ref, out)``, and the error message is ``(i, j, ref, out)``. It works, I admit. But it is rather confusing.